atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/async/CqlQuerySpec.java

```diff
@@ -14,10 +14,29 @@
  * limitations under the License.
  */
 
-package com.palantir.atlasdb.keyvalue.cassandra.async.query.forming;
+package com.palantir.atlasdb.keyvalue.cassandra.async;
+
+import java.util.function.Supplier;
+
+import org.immutables.value.Value;
 
 import com.palantir.atlasdb.keyvalue.api.TableReference;
 
-public interface QueryFormer {
-    String formQuery(SupportedQuery supportedQuery, String keySpace, TableReference tableReference);
+@Value.Immutable
+public interface CqlQuerySpec<R> {
+    @Value.Parameter
+    String keySpace();
+
+    @Value.Parameter
+    TableReference tableReference();
+
+    @Value.Parameter
+    SupportedQuery supportedQuery();
+
+    @Value.Parameter
+    Supplier<RowStreamAccumulator<R>> rowStreamAccumulator();
```

Comment:
`rowStreamAccumulatorFactory` since this gives you the supplier and not the actual accumulator

Reply:
Renamed.